solr-ingest/src/main/java/edu/unc/lib/dl/data/ingest/solr/filter/SetAccessStatusFilter.java

```diff
@@ -0,0 +1,99 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.data.ingest.solr.filter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import edu.unc.lib.dl.acl.fcrepo4.InheritedAclFactory;
+import edu.unc.lib.dl.acl.fcrepo4.ObjectAclFactory;
+import edu.unc.lib.dl.acl.service.PatronAccess;
+import edu.unc.lib.dl.acl.util.AccessPrincipalConstants;
+import edu.unc.lib.dl.data.ingest.solr.exception.IndexingException;
+import edu.unc.lib.dl.data.ingest.solr.indexing.DocumentIndexingPackage;
+import edu.unc.lib.dl.fcrepo4.ContentObject;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.search.solr.util.FacetConstants;
+/**
+ * Sets access-related status tags
+ *
+ * @author harring
+ *
+ */
+public class SetAccessStatusFilter implements IndexDocumentFilter {
+    private static final Logger log = LoggerFactory.getLogger(SetAccessStatusFilter.class);
+    private InheritedAclFactory inheritedAclFactory;
+    private ObjectAclFactory objAclFactory;
+
+    @Override
+    public void filter(DocumentIndexingPackage dip) throws IndexingException {
+        log.debug("Performing set access status filter on {}", dip.getPid());
+        dip.getDocument().setStatus(determineAccessStatus(dip));
+    }
+
+    /**
+     * Sets inherited acl factory
+     *
+     * @param aclFactory an inherited acl factory
+     */
+    public void setInheritedAclFactory(InheritedAclFactory iaf) {
+        this.inheritedAclFactory = iaf;
+    }
+
+    /**
+     * Sets non-inherited acl factory
+     *
+     * @param aclFactory an inherited acl factory
+     */
+    public void setObjectAclFactory(ObjectAclFactory oaf) {
+        this.objAclFactory = oaf;
+    }
+
+    private List<String> determineAccessStatus(DocumentIndexingPackage dip)
+            throws IndexingException {
+
+        ContentObject obj = dip.getContentObject();
+        PID pid = obj.getPid();
+        List<String> status = new ArrayList<>();
+
+        PatronAccess inheritedAccess = inheritedAclFactory.getPatronAccess(pid);
+        PatronAccess objAccess = objAclFactory.getPatronAccess(pid);
+
+        if (inheritedAclFactory.isMarkedForDeletion(pid)) {
+            status.add(FacetConstants.MARKED_FOR_DELETION);
+        }
+
+        if (inheritedAclFactory.getEmbargoUntil(pid) != null) {
+            status.add(FacetConstants.EMBARGOED);
+        }
+
+        if (inheritedAccess.equals(PatronAccess.none)) {
+            status.add(FacetConstants.STAFF_ONLY_ACCESS);
+        } else if (inheritedAclFactory.getPrincipalRoles(pid).containsKey(AccessPrincipalConstants.PUBLIC_PRINC)) {
+            status.add(FacetConstants.PUBLIC_ACCESS);
+        }
+
+        if (inheritedAccess.equals(PatronAccess.none) && !objAccess.equals(PatronAccess.none)) {
```

Comment:
👍 good that something can be marked STAFF_ONLY_ACCESS and PARENT_HAS_STAFF_ONLY_ACCESS